torch/csrc/jit/passes/quantization/insert_observers.cpp

```diff
@@ -1025,7 +1032,13 @@ InsertObserversHelper::insertObserversFor(
           propagateObservedProperty(v, block_observed_values);
           if (!inputs_outputs.count(v) &&
               !isObserved(v, block_observed_values)) {
-            if (auto observer_opt = getObserverFor(v)) {
+            auto observer_opt = getObserverFor(v);
+            // If the node is one of the propagate quant node, e.g.
```

Comment:
Isnt this logic also true for add or conv? Why special case concat?

Reply:
This is only true for cat, for conv we'll always quantize its input and output

Reply:
this is only true for cat, for conv we'll always quantize its input and output

Reply:
We need a consistent set of rules, consider the following two models:
``` python
import torch
import torch.nn as nn
class testM(nn.Module):
    def __init__(self):
        super().__init__()
        self.c = nn.Conv2d(3,5,1)
        self.d = nn.Conv2d(3,5,1)
    
    def forward(self, x):
# If there is a nn.Identity or shape, will the inputs be quantized?
        y = self.c(x)
        z = self.d(x)
        w = torch.cat((y, z))
        return w

# Second one:
class testM2(nn.Module):
    def __init__(self):
        super().__init__()
        self.c = nn.Conv2d(3,5,1)
        self.d = nn.Conv2d(3,5,1)
    
    def forward(self, x):
        w = torch.cat((x, x))
        y = self.c(w)
        z = self.d(w)
        return w

```
In the first case, the input will be quantized. In the second case the input will not be. 
What if in the first case I have an nn.Identity prior to the conv or a reshape? In that case do we quantize the inputs?

Reply:
we don't quantize identity or reshape, we also don't accept input that's already quantized outside of the model.

Reply:
In terms user interface, user will always provide a floating point Tensor, regardless of how the model is quantized

Reply:
Got it, the input is always in fp and in certain cases the input is quantized (conv) and in certain cases it is not (cat)